### PR TITLE
[Bug][ConstantPruningModifier] Fix mask de register bug

### PR DESCRIPTION
**Pull request by rahul-tuli**
This PR fixes a bug where, buffers were being de-registered when mask was persistent (this should be reversed)
Also fixes the logic for de-registration; `unregister_buffer` raises an `AttributeError` on torch Modules
Also updates test to check buffer was actually de-registered

### CODE REVIEW

File: tests/sparseml/modifiers/pruning/constant/test_pytorch.py [hunk 1]
```diff
@@ -68,7 +68,7 @@ def _test_optims():
 
     return [
         torch.optim.Adam,
-        torch.optim.SGD,
+        # torch.optim.SGD,
```
Comment (rahul-tuli) [inline tests/sparseml/modifiers/pruning/constant/test_pytorch.py:71]:
```suggestion
        torch.optim.SGD,
```

**Review by Satrat — APPROVED** [review]

**Review by bfineran — APPROVED** [review]
